src/compiler/checker.ts

```diff
@@ -7845,6 +7851,16 @@ namespace ts {
                 if (isBindingPattern(declaration.name)) {
                     return getTypeFromBindingPattern(<BindingPattern>declaration.name, /*includePatternInType*/ true);
                 }
+                if (isBindingPattern(declaration.parent)) {
+                    const parentDeclaration = declaration.parent.parent;
+                    const name = declaration.propertyName || declaration.name;
+                    if (isVariableLike(parentDeclaration) &&
+                        parentDeclaration.type &&
+                        (name.kind === SyntaxKind.Identifier || name.kind == SyntaxKind.StringLiteral)) {

```

Comment:
Yup, looks like ComputedPropertyName was missing. Thanks for the help finding and fixing the oversight.